lib/commands/backup.js

```diff
@@ -0,0 +1,200 @@
+'use strict';
+const Command = require('../command');
+const Promise = require('bluebird');
+const createDebug = require('debug');
+const Zip = require('adm-zip');
+const walker = require('klaw-sync');
+const fs = require('fs-extra');
+const path = require('path');
+const errors = require('../').errors;
+
+const debug = createDebug('ghost-cli:backup');
+
+function absolutePath(inputPath) {
+    return path.isAbsolute(inputPath) ? inputPath : path.join(process.cwd(), inputPath);
+}
+
+class BackupCommand extends Command {
+    run(argv) {
+        const instance = this.system.getInstance();
+
+        return this.ui.listr([{
+            task: this.initialize,
+            title: 'Checking configuration'
+        }, {
+            title: 'Backing up extensions',
+            task: this.backupExtensions
+        }, {
+            title: 'Backing up database',
+            task: this.backupDatabase
+        }, {
+            title: 'Backing up config files',
+            task: this.backupConfig
+        }, {
+            title: 'Backing up content folder',
+            task: this.backupContent
+        }, {
+            title: 'Writing backup file',
+            task: this.writeZipFile
+        }],{
+            env: this.system.environment,
```

Comment:
This causes an issue later on - because you run the `checkEnvironment` function below, the `system.environment` property has the potential to get updated, and therefore the value of `env` would be different.